src/main/views/viewManager.ts

```diff
@@ -100,13 +100,23 @@ export class ViewManager {
         if (ServerManager.hasServers()) {
             // TODO: This init should be happening elsewhere, future refactor will fix this
             ServerViewState.init();
-            await updateServerInfos(ServerManager.getAllServers());
             LoadingScreen.show();
-            ServerManager.getAllServers().forEach((server) => this.loadServer(server));
+
+            // We need to wait for the current server to be initialized before showing anything
+            // But we can initialize other servers in parallel
+            const otherServers = ServerManager.getAllServers().filter((server) => server.id !== ServerViewState.getCurrentServer().id);
+            const currentServer = ServerViewState.getCurrentServer();
+            otherServers.forEach((server) => this.initServer(server));
```

Comment:
Non-blocking nit: I have not tested this, but it seems simpler to me:
```suggestion
            otherServers.forEach(this.initServer);
```